src/chapters/thymeleaf-views/template-forms.rst

```diff
@@ -0,0 +1,152 @@
+   project ``coding-events``.
+#. Be sure to add the *Thymeleaf* dependency in addition to *Spring Web*, and
+   *Spring Boot DevTools*.
+
+   .. figure:: ./figures/codingEventsInit.png
+      :alt: Initialize ``coding-events`` project.
+      :scale: 80%
+
+#. *Generate* the ``.zip`` file and then import it into IntelliJ.
+
+Try It!
+--------
+
+Now that you have ``coding-events`` up and running, add features to it by
+coding along with the videos below:
+
+.. todo:: ``Coding Events App: Intro`` video goes here...
+
+.. todo:: ``Coding Events App: Render Add Event Form`` video goes here...
+
+A summary of Thymeleaf forms is given below, but remember that the text
+supports the videos and is NOT intended as a replacement.
+
+Create and Render a Form
+-------------------------
+
+A Thymeleaf form is simply a template that includes a ``<form>`` element inside
+the ``body`` of the HTML. The method for the form should be of type ``post``.
+
+.. sourcecode:: HTML
+   :linenos:
+
+   <body>
+
+      <!-- Other HTML -->
+
+      <form method="post">
+         <input type="text" name="inputName">
+         <input type="submit" value="submitButtonText">
+      </form>
+
+      <!-- Other HTML -->
+
+   </body>
+
+You can include as many inputs as you need in the form, and these can be of
+different types (e.g. text, email, checkbox, etc.). However, each different
+piece of data you want to collect needs to have a unique ``name`` attribute.
+
+To *render* the form in the view, add a method to the controller using the
+``@GetMapping`` annotation:
+
+.. sourcecode:: Java
+   :linenos:
+
+   @GetMapping("formTemplateName")
+   public String renderFormMethodName(Model model) {
+
+      // Method code...
+
+      return "pathToTemplate";
+   }
+
+Some points to note:
+
+#. Line 1: The string parameter for ``GetMapping`` must be the name of the form
+   template you want to use.
+#. Line 2: Declare a ``Model`` object to hold data that needs to be passed to
+   the template.
+#. The method code performs any data manipulation required before rendering the
+   form. The ``model.addAttribute`` statements would be included here.
+#. The ``return`` string specifies the path to the template. Recall that Spring
+   automatically adds MOST of the file path---up through ``.../templates``. You
+   need to add any path details that follow.
+
+   a. For example, if our ``templates`` folder contains a subfolder called
+      ``events`` that holds a template called ``create.html``, then line 6
+      would be ``return "events/create";``.
+
+Add a Form Handler Method
+--------------------------
+
+Now that you have created and rendered a form in your ``coding-events``
+project, you need to add a method to the controller to *handle* its submission.
+Code along with the video below to add this functionality.
+
+.. todo:: ``Coding Events App: Process Add Event Form`` video goes here...
+
+As usual, the following summary outlines the ideas from the clip.
```

Comment:
I would add more written support for this concept.

Reply:
At this point in time, I'm not sure what extra information should be added.  Suggestions?